src/Core/Managed/Shared/DataContracts/AggregatedMetricTelemetry.cs

```diff
@@ -0,0 +1,157 @@
+            this.StandardDeviation = standardDeviation;
+        }
+
+        /// <summary>
+        /// Gets or sets date and time when event was recorded.
+        /// </summary>
+        public DateTimeOffset Timestamp { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value that defines absolute order of the telemetry item.
+        /// </summary>
+        public string Sequence { get; set; }
+
+        /// <summary>
+        /// Gets the context associated with the current telemetry item.
+        /// </summary>
+        public TelemetryContext Context { get; }
+
+        /// <summary>
+        /// Gets or sets the name of the metric.
+        /// </summary>
+        public string Name
+        {
+            get { return this.Metric.name; }
+            set { this.Metric.name = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the number of samples for this metric.
+        /// </summary>
+        public int Count
+        {
+            get { return this.Metric.count ?? 0; }
+            set { this.Metric.count = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the value of this metric.
+        /// </summary>
+        public double Sum
+        {
+            get { return this.Metric.value; }
+            set { this.Metric.value = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the min value of this metric.
+        /// </summary>
+        public double Min
+        {
+            get { return this.Metric.min ?? 0; }
+            set { this.Metric.min = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the max value of this metric.
+        /// </summary>
+        public double Max
+        {
+            get { return this.Metric.max ?? 0; }
+            set { this.Metric.max = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the standard deviation of this metric.
+        /// </summary>
+        public double StandardDeviation
+        {
+            get { return this.Metric.stdDev ?? 0; }
+            set { this.Metric.stdDev = value; }
+        }
+
+        /// <summary>
+        /// Gets a dictionary of application-defined property names and values providing additional information about this metric.
+        /// </summary>
+        public IDictionary<string, string> Properties
+        {
+            get { return this.Data.properties; }
+        }
+
+        /// <summary>
+        /// Sanitizes the properties based on constraints.
+        /// </summary>
+        void ITelemetry.Sanitize()
+        {
+            this.Name = this.Name.SanitizeName();
+            this.Name = Utils.PopulateRequiredStringValue(this.Name, "name", typeof(AggregatedMetricTelemetry).FullName);
+            this.Properties.SanitizeProperties();
+            this.Count = this.Count < 0 ? 0 : this.Count;
+            this.Sum = Utils.SanitizeNanAndInfinity(this.Sum);
+            this.Min = Utils.SanitizeNanAndInfinity(this.Min);
```

Comment:
this call will initialize nullable field. Which is bad as we will need to pass the value over the wire even if we do not want to. Typical example is performance counter value. I do not want to spent bytes over the wire to send the `min` and `max` which are equal to the current value.